components/Header/index.js

```diff
@@ -1,16 +1,102 @@
 import styled from "styled-components";
+import { Spin as Hamburger } from "hamburger-react";
+import Link from "next/link";
+import { useState } from "react";
 
 export default function Header() {
-  return <StyledHeader>TrailBlazer</StyledHeader>;
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <StyledHeader>
+        <StyledHeading>TrailBlazer</StyledHeading>
+        <StyledMenu>
+          <Hamburger
+            toggled={isOpen}
+            toggle={setIsOpen}
+            duration={0.6}
+            label="Show Menu"
+          />
+        </StyledMenu>
+      </StyledHeader>
+      {isOpen ? (
+        <HamburgerMenu>
+          <MenuLink onClick={() => setIsOpen(false)} href="/bookmark">
+            My Bookmarks
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/">
+            Home
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/spotlight">
+            Inspire me!
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/createActivity">
+            Add new Activity
+          </MenuLink>
+        </HamburgerMenu>
+      ) : null}
+    </>
+  );
 }
 
-const StyledHeader = styled.h1`
+const StyledHeader = styled.div`
   background-color: var(--dijon);
-  padding: 10px 0;
   margin: 0;
-  border: 1px solid grey;
+  border: 1px solid black;
   position: fixed;
   width: 100vw;
   z-index: 3;
   box-shadow: var(--box-shadow-layout);
 `;
+
+const StyledHeading = styled.h1`
+  font-style: italic;
+  font-weight: 400;
+`;
+
+const StyledMenu = styled.div`
+  position: fixed;
+  position: absolute;
+  right: 20px;
+  top: 15px;
+  z-index: 10;
+
+  @media (max-width: 767px) {
+    display: none;
+  }
+`;
+
+const HamburgerMenu = styled.div`
```

Comment:
I would change the element to be a `nav` element, since it is more semantic and clearly indicates that the nested elements will be used to navigate through the application